scripts/dkim_registry.ts

```diff
@@ -0,0 +1,134 @@
+    });
+
+    const wallet = new ethers.Wallet(privateKey, provider);
+
+    const abi = [
+        "function setDKIMPublicKeyHash(string domainName, bytes32 publicKeyHash, address authorizer, bytes signature)",
+        "function revokeDKIMPublicKeyHash(string domainName, bytes32 publicKeyHash, address authorizer, bytes signature)"
+    ];
+    return new Contract(USEROVERRIDEABLE_DKIM_REGISTRY_ADDRESS, abi, wallet);
+}
+
+async function setDKIMPublicKeyHash(
+    domainName: string,
+    publicKeyHash: string,
+    authorizer: string,
+    signature: string,
+    privateKey: string
+) {
+    const registry = initializeContract(privateKey);
+
+    try {
+        const tx = await registry.setDKIMPublicKeyHash(domainName, publicKeyHash, authorizer, signature);
+        console.log("Transaction Hash:", tx.hash);
+        const receipt = await tx.wait();
+        if (receipt) {
+            console.log("Transaction was mined in block:", receipt.blockNumber);
+        } else {
+            console.log(`tx receipt not found`);
+        }
+    } catch (error) {
+        console.error("Error sending transaction:", error);
+    }
+}
+
+async function revokeDKIMPublicKeyHash(
+    domainName: string,
+    publicKeyHash: string,
+    authorizer: string,
+    signature: string,
+    privateKey: string
+) {
+    const registry = initializeContract(privateKey);
+
+    try {
+        const tx = await registry.revokeDKIMPublicKeyHash(domainName, publicKeyHash, authorizer, signature);
+        console.log("Transaction Hash:", tx.hash);
+        const receipt = await tx.wait();
+        if (receipt) {
+            console.log("Transaction was mined in block:", receipt.blockNumber);
+        } else {
+            console.log(`tx receipt not found`);
+        }
+    } catch (error) {
+        console.error("Error sending transaction:", error);
+    }
+}
+
+async function main() {
+    const program = new Command();
+
+    program
+        .option('-c, --command <type>', 'Command to execute: set or revoke')
+        .option('-p, --private-key <key>', 'Sender private key')
+        .option('-a, --authorizer-private-key <key>', 'Authorizer private key')
+        .parse(process.argv);
+
+    const options = program.opts();
+
+    const command = options.command;
+    const privateKey = options.privateKey;
+    const privateKeyForAuthorizer = options.authorizerPrivateKey;
+
+    if (!command || (command !== "set" && command !== "revoke")) {
+        console.error("Error: Please specify 'set' or 'revoke' as the command.");
+        process.exit(1);
+    }
+
+    if (!privateKey) {
+        console.error("Error: Please provide a private key.");
+        process.exit(1);
+    }
+
+    if (!privateKeyForAuthorizer) {
+        console.error("Error: Please provide a private key for the authorizer.");
+        process.exit(1);
+    }
+
+    const authorizerWallet = new ethers.Wallet(privateKeyForAuthorizer);
+    const authorizer = authorizerWallet.address;
```

Comment:
@wshino 
`authorizer` address should be also provided as the argument to the script.

Reply:
@SoraSuegami Updated